### ISSUE TITLE
[BUG]

### ISSUE THREAD

**Issue by ssbarnea** [willmcgugan/rich#346]
**Describe the bug**
When trying to install rich in editable mode user may get into error below. This is likely caused by current packaging configuration.

**To Reproduce**
```
$ pip install -e .
Obtaining file:///Users/ssbarnea/c/os/rich
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
    Preparing wheel metadata ... done
Requirement already satisfied: typing-extensions<4.0.0,>=3.7.4 in /Users/ssbarnea/.pyenv/versions/3.8.2/lib/python3.8/site-packages (from rich==7.0.0) (3.7.4.2)
Requirement already satisfied: pygments<3.0.0,>=2.6.0 in /Users/ssbarnea/.pyenv/versions/3.8.2/lib/python3.8/site-packages (from rich==7.0.0) (2.6.1)
Requirement already satisfied: colorama<0.5.0,>=0.4.0 in /Users/ssbarnea/.pyenv/versions/3.8.2/lib/python3.8/site-packages (from rich==7.0.0) (0.4.3)
Requirement already satisfied: commonmark<0.10.0,>=0.9.0 in /Users/ssbarnea/.pyenv/versions/3.8.2/lib/python3.8/site-packages (from rich==7.0.0) (0.9.1)
Installing collected packages: rich
  Running setup.py develop for rich
    ERROR: Command errored out with exit status 1:
     command: /Users/ssbarnea/.pyenv/versions/3.8.2/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/ssbarnea/c/os/rich/setup.py'"'"'; __file__='"'"'/Users/ssbarnea/c/os/rich/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps
         cwd: /Users/ssbarnea/c/os/rich/
    Complete output (3 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
    ModuleNotFoundError: No module named 'setuptools'
    ----------------------------------------
ERROR: Command errored out with exit status 1: /Users/ssbarnea/.pyenv/versions/3.8.2/bin/python3.8 -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/Users/ssbarnea/c/os/rich/setup.py'"'"'; __file__='"'"'/Users/ssbarnea/c/os/rich/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' develop --no-deps Check the logs for full command output.
noglob pip install -e .  5.32s user 0.79s system 98% cpu 6.225 total
```

**Platform**
Rich from master branch.
```
pip 20.2.3 from /Users/ssbarnea/.pyenv/versions/3.8.2/lib/python3.8/site-packages/pip (python 3.8)
```

**Comment by willmcgugan** [comment]
Use `poetry install`

**Comment by ssbarnea** [comment]
I checked and `poetry install` does not install rich in editable mode. In fact I do not even know what it does install as the package is not installed after:

```
ssbarnea@imac: ~/c/os/rich feature/fluidlogger ⚡
$ poetry install                                                                                                   Installing dependencies from lock file

No dependencies to install or update

  - Installing rich (8.0.0)

ssbarnea@imac: ~/c/os/rich feature/fluidlogger ⚡
$ python examples/log.py                                                                                           Traceback (most recent call last):
  File "examples/log.py", line 6, in <module>
    from rich.console import Console
ModuleNotFoundError: No module named 'rich.console'
FAIL: 1
```

While I do respect your option for poetry, I think that ignoring the ubiquitous pip may become a deal breaker for increasing the number of contributors (they will want editable installs) or even users (normal installs, ones that are working fine now). At this moment I have 16 packages installed in editable mode as I often contribute back to dependencies that I use tools or libraries.

My experience with poetry is quite low and after reading https://github.com/python-poetry/poetry/issues/34#issuecomment-600021371 thread I got even more confused about what it does and how.


For example reading output `poetry install --help` make me realise it does not claim to install any current package at all, it mentions what would do about dependencies but nothing about the current package/project itself. Is really poetry install a replacement for pip install?
